src/main/java/teammates/storage/api/InstructorsDb.java

```diff
@@ -54,6 +55,20 @@ public void putDocument(InstructorAttributes instructorParam) {
         }
     }
 
+    public void putDocuments(List<InstructorAttributes> instructorParams) {
+        List<SearchDocument> instructorDocuments = new ArrayList<SearchDocument>();
+        for (InstructorAttributes instructor : instructorParams) {
+            if (instructor.key == null) {
+                instructor = this.getInstructorForEmail(instructor.courseId, instructor.email);
+            }
+            // defensive coding for legacy data
+            if (instructor.key != null) {
+                instructorDocuments.add(new InstructorSearchDocument(instructor));
+            }
```

Comment:
It can be one of these two?

```java
if (instructor.key != null) {
    ...
} else {
    ...
}
```


```java
if (instructor.key != null) {
    ...
} 

// normal path; not indented

```